client/lib/upgrades/actions/domain-management.js

```diff
@@ -48,24 +47,18 @@ function setPrimaryDomain( siteId, domainName, onComplete = noop ) {
 			return onComplete( error, data );
 		}
 
-		sites.setSelectedSite( siteId );
+		dispatch( domainsPrimarySetCompletedAction( siteId, domainName ) );
```

Comment:
If we would return updated site object from the server as `data` then we could use existing action for site update.